homeassistant/components/huawei_lte/__init__.py

```diff
@@ -427,15 +439,29 @@ def service_handler(service) -> None:
             return
 
         if service.service == SERVICE_CLEAR_TRAFFIC_STATISTICS:
+            if router.suspended:
+                _LOGGER.debug("%s: ignored, integration suspended", service.service)
+                return
             result = router.client.monitoring.set_clear_traffic()
             _LOGGER.debug("%s: %s", service.service, result)
         elif service.service == SERVICE_REBOOT:
+            if router.suspended:
+                _LOGGER.debug("%s: ignored, integration suspended", service.service)
+                return
             result = router.client.device.reboot()
             _LOGGER.debug("%s: %s", service.service, result)
+        elif service.service == SERVICE_RESUME_INTEGRATION:
+            # Login will be handled automatically on demand
+            router.suspended = False
+            _LOGGER.debug("%s: %s", service.service, "done")
+        elif service.service == SERVICE_SUSPEND_INTEGRATION:
+            router.logout()
```

Comment:
Is it safe to logout if already logged out?

Reply:
it will cause an error (ResponseErrorLoginRequiredException) which is gracefully handled and logged in debug level, but I think it's better to do this just in case we've lost track/knowledge whether we're actually logged in or not.